content/spire/try/oidc-federation-aws.md

```diff
@@ -0,0 +1,415 @@
+---
+title: AWS OIDC Authentication
+description: Using OIDC to Authenticate SPIRE to AWS S3 on Kubernetes
+weight: 4
+toc: true
+aliases:
+menu:
+  spire:
+    weight: 50
+    parent: 'spire-try'
+---
+
+This tutorial builds on the [Kubernetes Quickstart](/spire/try/getting-started-k8s/) guide to describe how a SPIRE identified workload can, using a JWT-SVID, authenticate to Amazon AWS APIs, assume an AWS IAM role, and retrieve data from an AWS S3 bucket. This avoids the need to create and deploy AWS IAM credentials with the workload itself.
+
+In this tutorial you will learn how to:
+
+* Deploy the OIDC Discovery Provider Service
+* Create the required DNS A record to point to the Discovery Provider document
+* Create a sample AWS identity provider, policy, role, and S3 bucket
+* Test access to the S3 bucket
+
+## Prerequisites
+
+Note the following required accounts, prerequisites, and limitations before starting this tutorial:
+
+* You'll need access to the Kubernetes environment that you configured when going through [Kubernetes Quickstart](/spire/try/getting-started-k8s/) 
+* You'll need access to the AWS console to configure an identity provider, policy, role, and S3 bucket
+* This tutorial cannot run on Minikube as the Kubernetes environment must be network accessible by AWS
+* You'll need the ability to configure a DNS A record for the SPIRE OIDC Discovery Provider document location (see next section)
+
+### Required DNS A Record for the OIDC Discovery Provider IP Address
+
+The SPIRE OIDC Discovery Provider provides a URL to the location of the discovery document specified by the OIDC protocol. After starting the spire-oidc service you'll need to configure a DNS A record to point to the external IP address of that service. In the YAML files set up for this tutorial, replace MY\_DISCOVERY\_DOMAIN with the FQDN of the planned DNS hostname. The FQDN value for MY\_DISCOVERY\_DOMAIN does not need to correspond to an existing domain on your network. It is specific to the Kubernetes environment.
```

Comment:
This paragraph confused me a little, and the statement "The FQDN value for MY\_DISCOVERY\_DOMAIN does not need to correspond to an existing domain on your network." confused me specifically as it implies the DNS entry doesn't need to be valid, but the entry does need to be valid. I need to have registered a domain name somewhere and have configured it's name-servers to have pointed to a DNS service that includes this domain.

Maybe a simpler way of expressing all of this would be:

"As part of this tutorial you will need to register a public DNS record that will resolve to the public IP address of your Kubernetes cluster. This will require you or an administrator to have registered a domain name (eg. `yutani.com`) with a domain name registrar, have configured it's name-server to point to a DNS service, and to be able to create an A record for a new subdomain (eg. `oidc-endpoint.yutani.com`) in that DNS service. If you don't have a registered domain name or access to a DNS service, services like [Google Domains](https://domains.google.com) can help you set one up for a fee.

Reply:
Agreed, done.